prusti/src/callbacks.rs

```diff
@@ -127,8 +133,42 @@ impl prusti_rustc_interface::driver::Callbacks for PrustiCompilerCalls {
                 }
             }
             CrossCrateSpecs::import_export_cross_crate(&mut env, &mut def_spec);
-            if !config::no_verify() {
-                verify(env, def_spec);
+
+            // determine procedures that have to be verified.
+            let (annotated_procedures, types) = env.get_annotated_procedures_and_types();
+
+            if config::show_ide_info() && !config::no_verify() {
+                let compiler_info = compiler_info::IdeInfo::collect(&env, &annotated_procedures, &def_spec);
+                let out = serde_json::to_string(&compiler_info).unwrap();
+                PrustiError::message(format!("compilerInfo{out}"), DUMMY_SP.into())
+                    .emit(&env.diagnostic);
+            }
+
+            // collect and output Information used by IDE:
+
+            if !config::no_verify() && !config::skip_verification() {
+                if config::selective_verify().is_none() {
+                    let verification_task = VerificationTask {
+                        procedures: annotated_procedures,
+                        types,
+                    };
+                    verify(env, def_spec, verification_task);
+                } else {
+                    let target_def_path = config::selective_verify().unwrap();
```

Comment:
Consider if it's better to rewrite the branch to `if let Some(..) = config::selective_verify() { .. } else { .. }` to avoid the dependency between `is_none()` and `unwrap()`.